atlasdb-client/src/main/java/com/palantir/atlasdb/table/description/ColumnMetadataDescription.java

```diff
@@ -22,11 +22,15 @@
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Set;
+import javax.annotation.Nullable;
 import javax.annotation.concurrent.Immutable;
 
 @Immutable
 public class ColumnMetadataDescription {
+    @Nullable
     final ImmutableSet<NamedColumnDescription> namedColumns;
+
+    @Nullable
     final DynamicColumnDescription dynamicColumn;
 
     public ColumnMetadataDescription() {
```

Comment:
Separate from this PR this class has odd constructors that don't seem to hold the invariant that one field is null and one non-null indicating either dynamic or static columns. There should probably just be an interface or abstract base class and separate implementations for dynamic & static columns, but that might be more painful to graft onto the serDe at this point, but we should tighten up the preconditions for constructors.